NodeJS-server/controllers/news.controller.js

```diff
@@ -0,0 +1,129 @@
+const path = require('path');
+const fs = require('fs');
+const _ = require('lodash');
+const ValidationError = require('../errors/ValidationError.component')
+
+function getArticleById(req, res, next) {
+  fs.readFile('./mockData.json', 'utf8', (err, data) => {
+    if (err) next(err);
+  
+    let newData = JSON.parse(data);
+    if (_.isUndefined(newData[req.params.id])) {
+      next(new ValidationError(`article with id = ${req.params.id} is not found`));
+    } else {
+      res.json(newData[req.params.id]);
+    }
+  });
+}
+
+function getNewsBlock(req,res,next) {
+  fs.readFile('./mockData.json', 'utf8', (err, data) => {
+    if (err) next(err);
+
+    res.json(JSON.parse(data));
+  })
+}
+
+function putOneArticleById(req,res,next) {
+  fs.readFile('./mockData.json', 'utf8', (err, data) => {
+    if (err) next(err);
+    
+    let newData = JSON.parse(data);
+    if (_.isUndefined(newData[req.params.id])) {
+      newData[req.params.id] = req.body;
+      fs.writeFile('./mockData.json', JSON.stringify(newData, null, 2), (err, data) => {
+        if (err) next(err);
```

Comment:
This callback nesting is called callback hell :)
It can be avoided using Promises and / or async/await.